pkg/swarm/swarm.go

```diff
@@ -158,8 +163,14 @@ func (c *chunk) WithTagID(t uint32) Chunk {
 	return c
 }
 
-func (c *chunk) WithStamp(s []byte) Chunk {
-	c.stamp = s
+func (c *chunk) WithStamp(stamp Stamp) Chunk {
```

Comment:
comment

Reply:
added the comments to the interface level